backend/muster-roll-service/src/main/java/org/egov/service/EnrichmentService.java

```diff
@@ -0,0 +1,192 @@
+
+    @Autowired
+    private MusterRollServiceConfiguration config;
+
+    @Autowired
+    private IdGenRepository idGenRepository;
+
+    @Autowired
+    private MusterRollRepository musterRollRepository;
+
+
+    /**
+     * Enrich estimate muster roll
+     * @param musterRollRequest
+     *
+     */
+    public void enrichEstimateMusterRoll(MusterRollRequest musterRollRequest) {
+        RequestInfo requestInfo = musterRollRequest.getRequestInfo();
+        MusterRoll musterRoll = musterRollRequest.getMusterRoll();
+
+        //Audit details
+        AuditDetails auditDetails = musterRollServiceUtil.getAuditDetails(requestInfo.getUserInfo().getUuid(), musterRoll, true);
+        musterRoll.setAuditDetails(auditDetails);
+
+        //status
+        musterRoll.setStatus(Status.ACTIVE);
+
+    }
+
+    /**
+     * Enrich create muster roll
+     * @param musterRollRequest
+     *
+     */
+    public void enrichCreateMusterRoll(MusterRollRequest musterRollRequest) {
+        RequestInfo requestInfo = musterRollRequest.getRequestInfo();
+        MusterRoll musterRoll = musterRollRequest.getMusterRoll();
+
+        //Id
+        musterRoll.setId(UUID.randomUUID());
+
+        //Audit details
+        AuditDetails auditDetails = musterRollServiceUtil.getAuditDetails(requestInfo.getUserInfo().getUuid(), musterRoll, true);
+        musterRoll.setAuditDetails(auditDetails);
+
+        //musterRollNumber - Idgen
+        String rootTenantId = musterRoll.getTenantId().split("\\.")[0];
+        List<String> musterNumbers = getIdList(requestInfo, rootTenantId
+                , config.getIdgenMusterRollNumberName(), config.getIdgenMusterRollNumberFormat(), 1);
+        if (musterNumbers != null && !musterNumbers.isEmpty()) {
+            String musterRollNumber = musterNumbers.get(0);
+            musterRoll.setMusterRollNumber(musterRollNumber);
+        } else {
+            throw new CustomException("MUSTER_NUMBER_NOT_GENERATED","Muster number is not generated from Idgen service");
+        }
+
+        //status
+        musterRoll.setStatus(Status.ACTIVE);
+
+    }
+
+    /**
+     * Enrich update muster roll
+     * @param musterRollRequest
+     */
+    public void enrichUpdateMusterRoll(MusterRollRequest musterRollRequest, MusterRoll existingMusterRoll) {
+        RequestInfo requestInfo = musterRollRequest.getRequestInfo();
+        MusterRoll musterRoll = musterRollRequest.getMusterRoll();
+        Workflow workflow = musterRollRequest.getWorkflow();
+
+
+        if (workflow.getAction().equalsIgnoreCase("VERIFY")) {
+            //VERIFY action can modify the totalAttendance.
+            List<IndividualEntry> individualEntries = existingMusterRoll.getIndividualEntries();
+            List<IndividualEntry> modifiedIndividualEntries = musterRoll.getIndividualEntries();
+            for (IndividualEntry individualEntry : individualEntries) {
+                for (IndividualEntry modifiedIndividualEntry : modifiedIndividualEntries)  {
```

Comment:
The scenario is to iterate over two lists of objects whose order may not match. I tried different approaches but it was increasing the complexity. Since it was unavoidable went with nested for loops.